fastcrypto/src/serde_helpers.rs

```diff
@@ -118,3 +121,241 @@ impl<'de> DeserializeAs<'de, ed25519_consensus::Signature> for Ed25519Signature
             .map_err(to_custom_error::<'de, D, _>)
     }
 }
+
+///
+/// Serialization of internal types.
+///
+/// Every type's serialization should work as following:
+/// - Return base64 encoded bytes for is_human_readable() serializers.
+/// - Else, return the bytes as a fixed size array. This should be the canonical representation of
+///   the object (e.g., as defined in RFCs).
+///
+/// Types that do store a *cached* version of the serialized object should implement [ToFromBytes]
+/// and call [serialize_deserialize_with_to_from_bytes].
+///
+/// Types that do *not* store a cached version of the serialized object should implement
```

Comment:
Is there any way to merge these two options? Why should a developer know if cache exists?

Reply:
Cache as in having `pub bytes: OnceCell<Vec<u8>>` inside the type. The developer should know if its type is using that or not, and depending on that use `ToFromBytes` vs `ToFromByteArray`.

Reply:
@benr-ml interesting that caching is public api, and because of this indeed even external devs who just use the api can (in theory) know if it uses cache or not. Not a strong fan of this knowledge requirement, but 🤷